src/secex-genprotimgvm-scripts/genprotimg-script.sh

```diff
@@ -6,11 +6,19 @@ echo "Preparing for genprotimg-daemon"
 
 source="/build/genprotimg"
 destination="/genprotimg"
+pkey="usr/lib/coreos/ignition.asc"
+
+trap "rm -f ${source}/${pkey}" EXIT
 
 # Files need to be named correctly
 # genprotimg daemon can only see /genprotimg folder
 cp "${source}/vmlinuz" "${source}/initrd.img" "${source}/parmfile" "${destination}/"
 
+# Append Ignition GPG private key to initramfs
+cd "${source}"
+echo "${pkey}" | cpio --quiet -H newc -o | gzip -9 -n >> "${destination}/initrd.img"
+rm "${pkey}"
```

Comment:
I would but this part directly in `create_disk.sh` under `rdcore_replacement`. That way most of the logic stays out of the genprotimgvm, keeping it nice and simple, while keeping the logic where people look, inside `create_disk.sh`

Reply:
I did this by intention: `${destination}` is declared in this script and could be subject to change, so i decided not re-declare it within `create_disk.sh` .

Reply:
But as long as the key is inside of `initrd.img`, it get's copied over by the script. That way you would only need to add something along the lines of `trap rm -f "${source}/initrd.img"` to ensure that the key is deleted from the transport disk.

Reply:
idea is to give the `gentprotimg-script` all it needs and hide the `initrd` modification step. What's about moving the whole `rdcore_replacement` to a separate file and call similar to the call of `rdcore` (we can use similar cmdline options). This could be done as separate PR.
@jlebon wdyt?

Reply:
> What's about moving the whole `rdcore_replacement` to a separate file and call similar to the call of `rdcore` (we can use similar cmdline options). This could be done as separate PR. @jlebon wdyt?

Maybe. The idea of calling it like rdcore sounds appealing, but handling args in shell sucks. But also anyway ideally we'd dedupe the two paths more as discussed. So let's hold on that for now.

No strong opinion between doing the appending here or in `rdcore_replacement`.